syntax = "proto3";

message CreateRoomRequest{
}

message CreateRoomResponse{
  string roomnum = 3;
  string password = 4;
}

message ValidateRoomRequest{
  string roomnum = 1;
  string password = 2;
}

message ValidateRoomResponse{
  bool isSuccess = 1;
  string errcode = 2;
  string mssage = 3;
}

message JoinRoomResponse{
  bool isSuccess = 1;
  string errcode = 2;
  string mssage = 3;
}

message JoinRoomRequest{
  string userIp = 1;
  int32 userPort = 2;
  string roomnum = 3;
  string passwd = 4;
}

message GetRoomAddrRequest{
  string roomnum = 1;
}

message UserAddr{
  string ip = 1;
  int32 port = 2;
}

message GetRoomAddrResponse{
  repeated UserAddr addr = 1;
}

message LeaveRoomRequest{
  string userIp = 1;
  int32 port = 2;
}

message LeaveRoomResponse{
  bool isSuccess = 1;
  string errcode = 2;
  string mssage = 3;
}

message GetRoomNumByAddrRequest{
  string userIp = 1;
  int32 port = 2;
}

message GetRoomNumByAddrResponse{
  string roomnum = 1;
}

service Room{
  rpc Create(CreateRoomRequest) returns (CreateRoomResponse){}
  rpc Validate(ValidateRoomRequest) returns (ValidateRoomResponse){}
  rpc Join(JoinRoomRequest) returns (JoinRoomResponse){}
  rpc GetRoomAddr(GetRoomAddrRequest) returns (GetRoomAddrResponse){}
  rpc Leave(LeaveRoomRequest) returns (LeaveRoomResponse){}
  rpc GetRoomNumByAddr(GetRoomNumByAddrRequest) returns (GetRoomNumByAddrResponse){}
}

